// url-spinner@1.0.1/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var defaultOptions = {
    spinner: ["◓", "◑", "◒", "◐"],
    duration: 10000
};
function startSpinning(options) {
    if (!options)
        options = defaultOptions;
    else
        options = Object.assign({}, defaultOptions, options);
    var url = location.href;
    var render = makeRenderer(options.spinner);
    var timer = setInterval(render, 100, options);
    if (options.duration) {
        setTimeout(stopSpinning(timer, url), options.duration);
    }
    return stopSpinning(timer, url);
}
exports.startSpinning = startSpinning;
function makeRenderer(spinner) {
    var currentSpinnerIndex = 0;
    return function render() {
        window.history.replaceState(null, "", "?" + spinner[currentSpinnerIndex]);
        if (currentSpinnerIndex === spinner.length - 1) {
            currentSpinnerIndex = 0;
        }
        else {
            ++currentSpinnerIndex;
        }
    };
}
function stopSpinning(timer, url) {
    return function () {
        if (timer)
            clearInterval(timer);
        history.replaceState(null, "", url);
    };
}
